### PR TITLE
feat:encrypted_json_db

### PR DESCRIPTION
**Pull request by JarbasAl**
partially solves https://github.com/JarbasHiveMind/HiveMind-core/issues/14

only sqlite encryption to go

## Summary by CodeRabbit

- **New Features**
	- Introduced optional password handling for database connections, enhancing security for Redis and JSON storage options.
	- Added a new command-line option `--db-password` for various client management commands.

- **Bug Fixes**
	- Simplified error handling by removing conditional imports for SQLite.

- **Chores**
	- Updated requirements to include `pybase64` and a new dependency `json-database` (>= 0.8.1).

### CODE REVIEW

**Comment by JarbasAl** [comment]
needs rebase because of #28

**Comment by JarbasAl** [comment]
replaced by https://github.com/TigreGotico/json_database/pull/11 and #30

File: hivemind_core/database.py [hunk 1]
```diff
@@ -705,14 +709,20 @@ def __exit__(self, _type, value, traceback):
...
             "host": redis_host,
             "port": redis_port,
-            "password": redis_password
+            "password": db_password
+        })
+    elif db_backend == "json":
+        kwargs.update({
+            "name": db_name,
+            "subfolder": db_folder,
+            "password": db_password
```
[the remark this answers is not in the document]
Comment (goldyfruit) [reply hivemind_core/database.py:725]: Stale — the code has moved on. `database.py` in this fork contains no `redis` reference at all (`grep -c redis hivemind_core/database.py` -> 0), and no `in "redis"` anywhere in the tree. The substring trap you describe is real (`"red" in "redis"` and `"" in "redis"` are both True), but there is nothing left here to fix.

File: hivemind_core/scripts.py [hunk 2]
```diff
@@ -148,12 +146,11 @@ def rename_client(node_id, name,
               help="[json/sqlite] The name for the database file. ~/.cache/hivemind-core/{name}")
 @click.option("--db-folder", type=str, default="hivemind-core",
               help="[json/sqlite] The subfolder where database files are stored. ~/.cache/{db_folder}}")
+@click.option("--db-password", required=False, help="[json/redis] Password for accessing/decrypting the database (optional")
```
[the remark this answers is not in the document]
Comment (goldyfruit) [reply hivemind_core/scripts.py:149]: Stale for the same reason: `get_item_id` appears nowhere in the tree any more. Nothing to change.

File: hivemind_core/database.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (goldyfruit) [reply hivemind_core/database.py:725]: Apologies for the noise — I posted the comment above on the wrong repository. I was reviewing threads on a downstream fork and matched this PR by number. I have un-resolved the thread so it is back as you left it. Nothing here needs your attention.

File: hivemind_core/scripts.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (goldyfruit) [reply hivemind_core/scripts.py:149]: Same here — wrong repository, matched by PR number while reviewing a downstream fork. Un-resolved and back as you left it. Sorry for the noise.

**Pull request closed without merging**